kotlinx-coroutines-core/build.gradle.kts

```diff
@@ -84,13 +87,44 @@ kotlin {
     jvm {
         // For animal sniffer
         withJava()
-        compilations.create("benchmark") { associateWith(this@jvm.compilations.getByName("main")) }
     }
 }
 
-benchmark {
-    targets {
-        register("jvmBenchmark")
+private fun KotlinMultiplatformExtension.setupBenchmarkSourceSets(ss: NamedDomainObjectContainer<KotlinSourceSet>) {
+    // Forgive me, Father, for I have sinned.
+    // Really, that is needed to have benchmark sourcesets be the part of the project, not a separate project
+    @Suppress("UnusedVariable")
+    val benchmarkMain by ss.creating {
+        dependencies {
+            implementation("org.jetbrains.kotlinx:kotlinx-benchmark-runtime:${version("benchmarks")}")
+        }
+        kotlin.srcDir("benchmarks/main/kotlin")
+    }
+
+    ss.create("jvmBenchmark") {
```

Comment:
Where does the inconsistency with `val benchmarkMain` come from? Why not `val jvmBenchmark by creating {`?

Reply:
Done